### PR TITLE
[telequebec] Add support for Brightcove

### PR DESCRIPTION
**Pull request by patrickdessalle**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x ] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x ] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Telequebec now has two types of videos. Limelight and Brightcove. It used to only be Limelight. I've added support for Brightcove and make the download use the proper download based on the type of the source.

This also fixes issue 25833

### CODE REVIEW

**Comment by madmath** [comment]
What's the best way to get this reviewed and merged? I tried patching it and it works great.

File: youtube_dl/extractor/telequebec.py [hunk 1]
```diff
@@ -21,6 +23,13 @@ def _limelight_result(media_id):
             'ie_key': 'LimelightMedia',
         }
 
+    def _brightcove_result(self, brightcove_id):
+        return self.url_result(
```
Comment (dstftw) [inline youtube_dl/extractor/telequebec.py:27]: Must be `url_transparent`.

File: youtube_dl/extractor/telequebec.py [hunk 2]
```diff
@@ -21,6 +23,13 @@ def _limelight_result(media_id):
...
+            smuggle_url(
+                self.BRIGHTCOVE_URL_TEMPLATE % brightcove_id,
+                {'geo_countries': ['CA']}),
+            'BrightcoveNew', brightcove_id)
```
Comment (dstftw) [inline youtube_dl/extractor/telequebec.py:31]: DRY with `_limelight_result`.

File: youtube_dl/extractor/telequebec.py [hunk 3]
```diff
@@ -58,7 +67,10 @@ def _real_extract(self, url):
             'https://mnmedias.api.telequebec.tv/api/v2/media/' + media_id,
             media_id)['media']
 
-        info = self._limelight_result(media_data['streamInfo']['sourceId'])
+        if media_data['streamInfo']['source'] == 'Brightcove':
+            info = self._brightcove_result(media_data['streamInfo']['sourceId'])
```
Comment (dstftw) [inline youtube_dl/extractor/telequebec.py:71]: Extract `media_data['streamInfo']['sourceId']` into variable.

File: youtube_dl/extractor/telequebec.py [hunk 4]
```diff
@@ -58,7 +67,10 @@ def _real_extract(self, url):
             'https://mnmedias.api.telequebec.tv/api/v2/media/' + media_id,
             media_id)['media']
 
-        info = self._limelight_result(media_data['streamInfo']['sourceId'])
+        if media_data['streamInfo']['source'] == 'Brightcove':
```
Comment (dstftw) [inline youtube_dl/extractor/telequebec.py:70]: Must not fail if `media_data['streamInfo']['source']` not available.

**Review by dstftw — CHANGES_REQUESTED** [review]
Add a test for Brightcove.

**Pull request closed without merging**